src/pages/ProcessDetail.tsx

```diff
@@ -46,329 +45,252 @@ interface IProps extends RouteComponentProps<MatchParams>, WrappedComponentProps
-        let enrichedProcess = processInstance as ProcessWithDetails;
-
-        const stepUserInput: InputForm | undefined = enrichedProcess.form;
-        const tabs = stepUserInput ? ["user_input", "process"] : ["process"];
-        const selectedTab = stepUserInput ? "user_input" : "process";
-
-        this.setState({
-            process: { ...this.state.process, ...enrichedProcess },
-            stepUserInput: stepUserInput,
-            tabs: tabs,
-            selectedTab: selectedTab,
-        });
-
-        if (this.state.autoScrollToLast) {
-            this.scrollToLast();
-        }
-
-        if (enrichedProcess.status === "completed" && this.state.httpIntervalFallback) {
-            clearInterval(this.state.httpIntervalFallback);
-        }
-    };
-
-    handleWebsocketError = (error: any) => {
-        setFlash(error.detail, "error");
-    };
-
-    httpfallback = () => {
-        if (this.state?.process?.status !== "completed" && !this.state.httpIntervalFallback) {
-            const httpIntervalFallback = setInterval(() => {
-                this.context.apiClient.process(this.props.match.params.id).then(this.updateProcessHttp);
-            }, 3000);
-            this.setState({ httpIntervalFallback });
-        }
-    };
-
-    componentDidMount = () => {
-        this.context.apiClient.process(this.props.match.params.id).then(this.initializeProcessDetails);
-    };
-
-    componentDidUpdate = () => {
-        const el = this.actionsRef.current;
-        if (this.state.observer || !this.actionsRef.current) {
+    useEffect(() => {
+        const el = actionsRef.current;
+        if (observer || !actionsRef.current) {
             return;
         }
 
-        const observer = new IntersectionObserver(([entry]) => {
-            this.setState({ isActionsMenuOnScreen: entry.isIntersecting });
+        const localObserver = new IntersectionObserver(([entry]) => {
+            setIsActionsMenuOnScreen(entry.isIntersecting);
         });
         if (el) {
-            observer.observe(el);
+            localObserver.observe(el);
         }
-        this.setState({ observer: observer });
-    };
+        setObserver(localObserver);
+    }, [observer]);
 
-    componentWillUnmount = () => {
-        if (this.state.observer) {
-            this.state.observer.disconnect();
-        }
-    };
+    useEffect(() => {
+        return () => {
+            if (observer) {
+                observer.disconnect();
+            }
+        };
+    }, [observer]);
 
-    handleUpdateProcess = (runningProcesses: WsProcessV2[]) => {
-        const process = runningProcesses.find((p) => p.pid === this.props.match.params.id);
-        if (this.state.wsProcess === process || !process) {
+    const handleUpdateProcess = (runningProcesses: WsProcessV2[]) => {
+        const localProcess = runningProcesses.find((p) => p.pid === props.match.params.id);
+        if (wsProcess === localProcess || !localProcess) {
             return <></>;
```

Comment:
You don't need to return a fragment anymore
```suggestion
            return;
```